src/FieldWarning/Assets/Units/Component/Movement/Pathfinder.cs

```diff
@@ -74,12 +80,58 @@ public class Pathfinder
 
         public Pathfinder(MovementComponent unit, PathfinderData data)
         {
+            
             _unit = unit;
             Data = data;
             _path = new List<PathNode>();
             FinalCompletionDist =
                     2f * Constants.MAP_SCALE + unit.Data.MinTurnRadius;
             _nextUpdateTime = 0f;
+
+            Thread calculationThread = new Thread(() => PathFinderRunner());
+            calculationThread.IsBackground = true;
+            calculationThread.Start();
```

Comment:
I'm a bit confused by the whole concept here. We only start one thread per pathfinder object. So why the semaphore at all? It's only taken in PathFinderRunner and that method is only called by that thread. What state are we protecting from concurrent access?

Also a minor nitpick: We start an entire thread per pathfinder object. So if we have 200 units we'd have 200 threads (assuming they have only one movement component each). In the future we'd want to make a thread pool or something. I'm not familiar enough with the code to easily make the change but I've been thinking for a while that the pathfinder should be a shared object (one per movement type?) that doesn't hold the paths for the movement components but rather is queried by them, gives them the path and then moves on to the next request.

Reply:
Yep for now its 1 thread per unit. We will need to change this later on if it causes issues by either using a thread pool or one thread for pathfinding.